### PR TITLE
feat: [DATA-2523] Added hyperopt helm chart

### PR DESCRIPTION
**Pull request by arihantsurana**
What changes are proposed in this pull request?
- Initialised the entire chart
- Used env vars from the examples in hyperkops repo
- Added a PR Template

How was this change tested?
- Tested using:
`kubectl config use-context docker-for-desktop`
`helm init`
`helm install .`

output:
```
$ helm install .   

NAME:   silly-newt
LAST DEPLOYED: Tue Jun 25 10:50:15 2019
NAMESPACE: default
STATUS: DEPLOYED

RESOURCES:
==> v1/ServiceAccount
NAME                               AGE
silly-newt-chart-hyperopt-monitor  4s

==> v1/Role
silly-newt-chart-hyperopt-monitor  4s

==> v1beta1/RoleBinding
silly-newt-chart-hyperopt-monitor  4s

==> v1/Service
silly-newt-chart-hyperopt-mongo  4s

==> v1beta2/Deployment
silly-newt-chart-hyperopt-mongo     4s
silly-newt-chart-hyperopt-monitor   4s
silly-newt-chart-hyperopt-workers   4s
silly-newt-chart-hyperopt-workload  3s

==> v1/Pod(related)

NAME                                                 READY  STATUS             RESTARTS  AGE
silly-newt-chart-hyperopt-mongo-698fcb9bb4-jvkd9     0/1    ContainerCreating  0         4s
silly-newt-chart-hyperopt-monitor-5d578c86f7-5gz5v   0/1    ContainerCreating  0         3s
silly-newt-chart-hyperopt-workers-6f55dfb844-7cdd8   0/1    ContainerCreating  0         3s
silly-newt-chart-hyperopt-workers-6f55dfb844-g6h5w   0/1    ContainerCreating  0         3s
silly-newt-chart-hyperopt-workload-64558fc6c4-4v9g5  0/1    ContainerCreating  0         3s

==> v1/Secret

NAME                       AGE
silly-newt-chart-hyperopt  4s


NOTES:
1. To connect mongo db using your local to deploy hyperopt job

export POD_NAME=$(kubectl get pods --namespace default -l "app.kubernetes.io/name=chart-hyperopt-mongo,app.kubernetes.io/instance=silly-newt" -o jsonpath="{.items[0].metadata.name}")

kubectl port-forward $POD_NAME 27017:27017

You can then connect to mongo db at:
localhost:27017

2. you can connect to the mongo db on other deployments within the kubernetes environment using this service url:
"silly-newt-chart-hyperopt-mongo.default.svc.cluster.local"
```

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,61 @@
 # chart-hyperopt
 © Hipages Group Pty Ltd 2019
 
-Helm Chart to deploy [hyperopt](https://github.com/hyperopt/hyperopt) on kubernetes
+Helm Chart to deploy [hyperopt](https://github.com/hyperopt/hyperopt): a Distributed Asynchronous Hyperparameter Optimization in Python
+This complements the [hyperkops](https://github.com/hipagesgroup/hyperkops) repository reared towards running hyperopt in Kubernetes.
```
Comment (datascienceChris) [inline README.md]:
```suggestion
This complements the [hyperkops](https://github.com/hipagesgroup/hyperkops) repository which enables the deployment of hyperopt in Kubernetes. 
```

File: values.yaml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    #  memory: 128Mi
+
+  nodeSelector: {}
+
+  tolerations: []
+
+  affinity: {}
+
+  env:
+    TIMEOUT_INTERVAL: 15
```
Comment (datascienceChris) [inline values.yaml]: remove the timeout interval

Comment (datascienceChris) [reply values.yaml]: These are the arguments (and defaults) required for the monitor:


mongo_db_address | MONGO_DB_ADDRESS | url to MongoDB | localhost
mongo_db_port | MONGO_DB_PORT | mongo db port | 27017
trials_db | TRIALS_DB | Name of the MongoDB in which the trials are stored | model_db
trials_collection | TRIALS_COLLECTION | Name of the Mongo Collection in which the trials are stored | jobs
update_interval | UPDATE_INTERVAL | Time between queries to the MongoDB to find failed jobs | 100
namespace | NAMESPACE | Namespace in which the Pods to be monitored are being deployed |  
label-selector | LABEL_SELECTOR | Labels which identify relevant Hyperkops worker Pods

Comment (datascienceChris) [reply values.yaml]: https://github.com/hipagesgroup/hyperkops/blob/DATA-2526/examples/kube-deploy-hyperkops-infrastructure.yaml
this might help

**Comment by arihantsurana** [comment]
Fixed the mongodb issue
image

File: Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+appVersion: "1.0"
+description: A Helm chart for Kubernetes
```
Comment (datascienceChris) [inline Chart.yaml]:
```suggestion
description: A Helm chart for deploying hyperkops - a technology for enabling hyperopt to be distributed and scaled on Kubernetes 
```

**Pull request merged**